### ISSUE TITLE
libreadline history shortcuts

### ISSUE THREAD

**Issue by mahrud** [Macaulay2/M2-emacs#43]
When using a terminal, or M2 inside a terminal, you can go back in command history with up/down arrow keys, but also Ctrl + P and Ctrl + N (and Ctrl + R for reverse search).

On Emacs, only Ctrl + arrow keys works, which in particular conflicts with a macOS shortcut Ctrl + Up (Ctrl + R works for different reasons).

We should probably also enable scrolling through history with Ctrl + P/N within M2-mode (currently it just moves up/down within the buffer, which you can always use the arrow keys for, but not through the history).

---

On a related note, we should really enable persistent history across M2 instances run from a terminal. This is pretty common in repl shells and makes many lives easier.

**Comment by mahrud** [comment]
Update: I learned that Meta + P/N work in Emacs, but they don't work in the terminal. The discrepancy is the annoying part, but I will check if this works on macOS.

**Comment by d-torrance** [comment]
I'd be very hesitant to have a major mode rebind something as basic as `C-p` and `C-n` -- these are very standard Emacs keybindings.

**Comment by mahrud** [comment]
Fair enough. Given that Meta keys work, I don't think switching C-P/N in Emacs worth it, so I changed the topic and content of the issue to be able changing M2 within terminal to align with Emacs shortcuts instead.
